File: app/src/main/java/app/komatatsu/todo/AddActivity.java
package app.komatatsu.todo;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class AddActivity extends AppCompatActivity {

    public static Intent getIntent(Context context) {
        return new Intent(context, AddActivity.class);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add);
        final EditText editText = findViewById(R.id.edit);
        Button button = findViewById(R.id.add);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ToDo todo = new ToDo(
                        System.currentTimeMillis(),
                        editText.getText().toString(),
                        false
                );
                TodoRepository.addItem(AddActivity.this, todo);
                finish();
            }
        });
    }
}

File: app/src/main/java/app/komatatsu/todo/AddActivity.kt
package app.komatatsu.todo

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.EditText

class AddActivity : AppCompatActivity() {

    companion object {
        @JvmStatic
        fun getIntent(context: Context): Intent {
            return Intent(context, AddActivity::class.java)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add)

        val title = findViewById<EditText>(R.id.edit)
        val add = findViewById<Button>(R.id.add)
        add.setOnClickListener {
            val todo = ToDo(
                    System.currentTimeMillis(),
                    title.text.toString(),
                    false
            )
            TodoRepository.addItem(this@AddActivity, todo)
            finish()
        }
    }
}
